src/TenancyMailManager.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy; // todo new Overrides namespace?
+
+use Illuminate\Mail\MailManager;
+use Stancl\Tenancy\Bootstrappers\MailTenancyBootstrapper;
+
+/**
+ * This class (paired with the MailTenancyBootstrapper)
+ * allows tenants to have their custom mailer credentials.
+ *
+ * Tenancy swaps Laravel's MailManager singleton for an instance of this class,
+ * which overrides the manager's get method to always resolve
+ * the mailers specified in the static $tenantMailers property
+ * instead of getting them from the $mailers property where they're cached.
+ *
+ * @see MailTenancyBootstrapper
+ */
+class TenancyMailManager extends MailManager
+{
+    /**
+     * Mailers to always resolve from the container (even when they're
+     * cached and available in the $mailers property).
+     */
+    public static array $tenantMailers = [];
+
+    /**
+     * Override the get method so that the mailers in $tenantMailers always get resolved,
+     * even when they're cached and available in the $mailers property.
+     */
+    protected function get($name)
+    {
+        if (in_array($name, static::$tenantMailers)) {
+            return $this->resolve($name);
+        }
+
+        return parent::get($name);
+    }
```

Comment:
I deleted the `smtp` default from `$tenantMailers` because if someone's app is only using e.g. `ses`, having `smtp` explicitly named as a tenant mailer by default wouldn't make much sense.

This made me think that the re-resolving could be done for all mailers by default, and to utilize caching of some mailers, we could be naming them explicitly in a `$mailersToGetFromCache` array. So it'd be the other way around – mailers in the array would not always get resolved, and the `get` method would look like this:
```php
if (in_array($name, static::$mailersToGetFromCache)) {
    return parent::get($name);
}

return $this->resolve($name);
```
(swap `return` in the `if` check with the other one)

@stancl, which approach seems better to you?

Reply:
I think it's better to specify the mailers that *should* be re-resolved.